sympy/calculus/util.py

```diff
@@ -481,7 +496,11 @@ def _check(orig_f, period):
             # checked below
 
     if isinstance(f, exp):
-        if im(f) != 0:
+
+        if re(f).has(f) or im(f).has(f): # Avoid infinite loop
+            period = periodicity(f.args[0], symbol)
```

Comment:
What is the effect of this? Here f is in instance of exp. We check if `re(f).has(f)` but what does that mean or why is it needed?

Reply:
If one defined a periodic function ``f`` (see test_util.py for example), ``re(exp(f(x)))`` gives ``exp(re(f(x))*cos(im(f(x)))``, and ``im(exp(f(x)))`` gives  ``exp(re(f(x))*cos(im(g(x)))``.  
I can't exactly recall, but I remember that this was causing infinite loop and so I added this ``if`` condition to avoid it.